### PR TITLE
next rev of opentracing spec

### PR DESCRIPTION
**Pull request by Steve-Fontes**
Signed-off-by: Steve Fontes <steve.m.fontes@gmail.com>

### CODE REVIEW

File: spec/src/main/asciidoc/architecture.asciidoc [hunk 1]
```diff
@@ -17,37 +17,144 @@
...
+([Examples](http://opentracing.io/documentation/pages/supported-tracers.html)).
+The storage service for distributed trace records can provide features to view
+the cross service trace records associated with particular request flows.
 
-Distributed tracing allows you to trace the flow of a request across service boundaries.
-This is particularly important in a microservices environment where a request typically flows through multiple services.
-To accomplish distributed tracing, each service must be instrumented to log messages with a correlation id that may have been propagated from an upstream service.
-A common companion to distributed trace logging is a service where the distributed trace records can be stored. ([Examples](http://opentracing.io/documentation/pages/supported-tracers.html)).
-The storage service for distributed trace records can provide features to view the cross service trace records associated with particular request flows.
+Opentracing.io is an API specification for working with a distributed tracing
```
Comment (pavolloffay) [inline spec/src/main/asciidoc/architecture.asciidoc]: Why domain name? 

I think `OpenTracing` is better

File: spec/src/main/asciidoc/architecture.asciidoc [hunk 2]
```diff
@@ -17,37 +17,144 @@
...
+system. Opentracing.io defines a Span as the conceptual artifact that is
+propagated between services. The Span information can also be communicated to a
+central server for storage.
 
-Opentracing.io is an API specification for working with a distributed tracing system. Opentacing.io defines a Span as the conceptual artifact that is propagated between services. The Span information can also be communicated to a central server for storage.
+The microprofile.io Opentracing proposal will describe how a developer can
```
Comment (pavolloffay) [inline spec/src/main/asciidoc/architecture.asciidoc]: s/Openracing/OpenTracing, possibly applies to more places

File: spec/src/main/asciidoc/architecture.asciidoc [hunk 3]
```diff
@@ -17,37 +17,144 @@
...
 
-Two annotations are provided to define how Spans can be used.
+=== Enabling distributed tracing with no code instrumentation
 
-* Trace: Specify a class or method to be traced.
+The microprofile.io implementation will allow JAX-RS applications to participate in distributed tracing, without requiring developers to add any distributed tracing code into their applications, and without requiring developers to know anything about the distributed tracing environment that their JAX-RS application will be deployed into.
 
-* NoTrace: Specify that a method is not to be traced.
+1. The microprofile.io implementation must provide a mechanism to configure an io.opentracing.Tracer implementation for use by each JAX-RS application.
+2. The microprofile.io implementation must provide a mechanism to automatically extract Span information from any incoming JAX-RS request.
```
Comment (pavolloffay) [inline spec/src/main/asciidoc/architecture.asciidoc]: This is usually called Context or SpanContext information

File: spec/src/main/asciidoc/architecture.asciidoc [hunk 4]
```diff
@@ -17,37 +17,144 @@
...
+3. The microprofile.io implementation must provide a mechanism to automatically start a Span for any incoming JAX-RS request.
+5. The microprofile.io implementation must provide a mechanism to automatically inject Span information into any outgoing JAX-RS request.
```
Comment (pavolloffay) [inline spec/src/main/asciidoc/architecture.asciidoc]: the same as above

File: spec/src/main/asciidoc/architecture.asciidoc [hunk 5]
```diff
@@ -17,37 +17,144 @@
...
+Spans created for incoming requests will have the following tags added by default:
+
+* Tags.SPAN_KIND = Server
+* Tags.HTTP_METHOD
+* Tags.HTTP_URL
+* Tags.HTTP_STATUS
+* Tags.ERROR (if true)
+
+==== Span creation and injection for outbound requests
+When a request is sent from a JAX-RS  javax.ws.rs.client.Client, a new Span will be created that is injected in the outbound request for propagation downstream. The new Span will be a child of the current Span if a current Span exists. The new Span will be finished when the outbound request is completed.
```
Comment (pavolloffay) [inline spec/src/main/asciidoc/architecture.asciidoc]: Some http client instrumentations go a little bit beyond this:

One span which wraps the whole invocation and N spans which are descendants of the first span to model each network attempt (e.g. redirects..). It's a little bit more verbose, but provides a better understanding.

File: spec/src/main/asciidoc/architecture.asciidoc [hunk 6]
```diff
@@ -17,37 +17,144 @@
 // limitations under the License.
 // Contributors:
 // Steve Fontes
-[[architecture]]
 
 == Architecture
 
-This specification defines an easy way to instrument an application to take advantage of an opentracing.io Tracer implementation.
+This specification defines an easy way to allow an application running in a
```
Comment (pavolloffay) [inline spec/src/main/asciidoc/architecture.asciidoc]: I haven't read it all.

Does the spec define a way to use span decorators?

File: spec/src/main/asciidoc/architecture.asciidoc (hunk 1 above)
Comment (Steve-Fontes) [reply spec/src/main/asciidoc/architecture.asciidoc]: Updated the spec

File: spec/src/main/asciidoc/architecture.asciidoc (hunk 2 above)
Comment (Steve-Fontes) [reply spec/src/main/asciidoc/architecture.asciidoc]: done

File: spec/src/main/asciidoc/architecture.asciidoc (hunk 3 above)
Comment (Steve-Fontes) [reply spec/src/main/asciidoc/architecture.asciidoc]: Added more info to spec

File: spec/src/main/asciidoc/architecture.asciidoc (hunk 6 above)
Comment (Steve-Fontes) [reply spec/src/main/asciidoc/architecture.asciidoc]: The spec defines default tags that will be added to created Spans. For other decorations, the programmer would have to do something like:
@Inject Tracer tracer;
....
tracer.activeSpan().setTag(...);

File: spec/src/main/asciidoc/architecture.asciidoc (hunk 5 above)
Comment (Steve-Fontes) [reply spec/src/main/asciidoc/architecture.asciidoc]: Can you point me to a particular client implementation to look at?

**Pull request merged**